src/ListGroupItem.js

```diff
@@ -1,100 +1,48 @@
-import React, { cloneElement } from 'react';
+import React from 'react';
+import OldListGroupItem from './deprecated/ListGroupItem';
 import bootstrapUtils, { bsStyles, bsClass } from './utils/bootstrapUtils';
+import deprecated from 'react-prop-types/lib/deprecated';
+
 import { State } from './styleMaps';
 import classNames from 'classnames';
 
 class ListGroupItem extends React.Component {
 
-
   render() {
-    let classes = bootstrapUtils.getClassSet(this.props);
+    let { componentClass, header } = this.props;
 
-    classes.active = this.props.active;
-    classes.disabled = this.props.disabled;
-
-    if (this.props.href) {
-      return this.renderAnchor(classes);
-    } else if (this.props.onClick) {
-      return this.renderButton(classes);
-    } else if (this.props.listItem) {
-      return this.renderLi(classes);
+    if (header) {
+      return <OldListGroupItem {...this.props} />;
     }
 
-    return this.renderSpan(classes);
-  }
-
-  renderLi(classes) {
-    return (
-      <li
-        {...this.props} className={classNames(this.props.className, classes)}>
-        {this.props.header ? this.renderStructuredContent() : this.props.children}
-      </li>
-    );
-  }
-
-  renderAnchor(classes) {
-    return (
-      <a
-        {...this.props}
-        className={classNames(this.props.className, classes)}
-      >
-        {this.props.header ? this.renderStructuredContent() : this.props.children}
-      </a>
-    );
-  }
+    let classes = bootstrapUtils.getClassSet(this.props);
 
-  renderButton(classes) {
-    return (
-      <button
-        type="button"
-        {...this.props}
-        className={classNames(this.props.className, classes)}>
-        {this.props.header ? this.renderStructuredContent() : this.props.children}
-      </button>
-    );
-  }
+    classes.active = this.props.active;
+    classes.disabled = this.props.disabled;
 
-  renderSpan(classes) {
-    return (
-      <span
-        {...this.props} className={classNames(this.props.className, classes)}>
-        {this.props.header ? this.renderStructuredContent() : this.props.children}
-      </span>
-    );
-  }
+    let className = classNames(this.props.className, classes);
 
-  renderStructuredContent() {
-    let header;
-    let headingClass = bootstrapUtils.prefix(this.props, 'heading');
+    const componentClasses = {
```

Comment:
The component class should be the 1st arg to `createElement` directly. See other instances of `componentClass` in our code.